subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GroovyCompileScriptBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.groovy.scripts
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.internal.scripts.CompileScriptBuildOperationType
+import org.gradle.internal.scripts.ScriptCompileStage
+
+class GroovyCompileScriptBuildOperationIntegrationTest extends AbstractIntegrationSpec {
+
+    def operations = new BuildOperationsFixture(executer, temporaryFolder)
+
+    def setup() {
+        executer.requireOwnGradleUserHomeDir()
+    }
+
+    def "captures script compilation build operations"() {
+        given:
+        settingsFile << "println 'settings.gradle'"
+        buildFile << """
+            apply from: 'script.gradle'
+            println 'build.gradle'
+        """
+
+        file("script.gradle") << "println 'script.gradle'"
+        file("init.gradle") << "println 'init.gradle'"
+
+        when:
+        succeeds 'help', '-I', 'init.gradle'
+
+        then:
+        def scriptCompiles = operations.all(CompileScriptBuildOperationType)
+
+        scriptCompiles*.displayName == [
+            'Compile script init.gradle (CLASSPATH)',
```

Comment:
❓ It feels wrong that there's an operation for `CLASSPATH` stage compilation for a script with no classpath block.

Reply:
it always happens. I guess because you cannot know upfront if the blocks you're looking for actually exist or not

Reply:
It'll be different with the Kotlin DSL where compilation only happen when necessary.

Reply:
still not faster ;)